src/Models/Test/ModelInstantiatorTest.php

```diff
@@ -292,7 +295,77 @@ public function testDataObjectToArray(): void
                 'birthDate'    => '2020-01-01T00:00:00+00:00',
                 'testProperty' => true,
             ],
-            $this->model_instantiator->dataObjectToArray($data_object)
+            $result
+        );
+    }
+
+    public function testDataObjectToArrayWithPrivacyProtectedDataObject(): void
+    {
+        // Create an input Privacy Protected Data Object instance.
+        $data_object = $this->getPrivacyProtectedTestDataObject();
```

Comment:
Perhaps the tests should verify that the private properties exist on the data object prior to the array transformation?

Reply:
Sure. I'll add an explicit assertion.